odoo/addons/test_lint/tests/_odoo_checker_sql_injection.py

```diff
@@ -34,41 +34,50 @@ def _get_cursor_name(self, node):
         cursor_name = '.'.join(expr_list)
         return cursor_name
 
-    def _get_func_name(self, node):
-        func_name = (
-            isinstance(node, astroid.Name) and node.name or
-            isinstance(node, astroid.Attribute) and node.attrname or '')
-        return func_name
+    def _allowable(self, node):
+        """
+        :type node: NodeNG
+        """
+        if isinstance(node, astroid.Call):
+            node = node.func
+        # self._thing is OK (mostly self._table), self._thing() also because
+        # it's a common pattern of reports (self._select, self._group_by, ...)
+        return (isinstance(node, astroid.Attribute)
+            and isinstance(node.expr, astroid.Name)
+            and node.attrname.startswith('_')
+        )
 
     def _check_concatenation(self, node):
-        is_bin_op = False
         if isinstance(node, astroid.BinOp) and node.op in ('%', '+'):
-            # execute("..." % self._table)
-            if (isinstance(node.right, astroid.Attribute) and
-                    not node.right.attrname.startswith('_')):
-                is_bin_op = True
             if isinstance(node.right, astroid.Tuple):
```

Comment:
List apply the same `node.right.elts`

```suggestion
            if isinstance(node.right, (astroid.Tuple, astroid.List)):
```

Reply:
Thinking about it more and testing in a python shell, that's not the case, `%` treats a list as a single parameter not as a sequence thereof.

Reply:
You are right!